lib/api/swagger.js

```diff
@@ -36,97 +38,115 @@ module.exports = function generateSwagger (kuzzle) {
+        description: 'Kuzzle Base Url',
+        variables: {
+          baseUrl: {
+            default: 'localhost'
+          },
+          port: {
+            default: '7512'
+          }
+        }
+
+      }
     ],
-    swagger: '2.0'
+    tags: [],
+    paths: {},
   };
 
   routes.forEach(route => {
-    // conforms to the swagger format by replacing the parameters notation :parameter by {parameter}
+    // Set :param notation to {param}
     route.url_ = route.url.replace(routeUrlMatch, '{$1}');
 
     if (swagger.paths[route.url_] === undefined) {
       swagger.paths[route.url_] = {};
     }
 
-    if (swagger.paths[route.url_][route.verb] === undefined) {
-
-      if (route.infos === undefined) {
-        route.infos = {};
-      }
+    if (swagger.paths[route.url_][route.verb] !== undefined) {
+      return;
+    }
 
-      if (route.infos.description === undefined) {
-        route.infos.description = `Controller: ${route.controller}. Action: ${route.action}.`;
-      }
-      else {
-        route.infos.description += `\nController: ${route.controller}. Action: ${route.action}.`;
+    if (route.info === undefined) {
+      route.info = {};
+    }
+    if (route.controller !== undefined) {
+      if (!_.some(swagger.tags, {name: route.controller})) {
+        const capitalizedController = route.controller.charAt(0).toUpperCase() + route.controller.slice(1);
+        swagger.tags.push({name: route.controller, description: `${capitalizedController} Controller`});
       }
-
-      if (route.infos.produces === undefined) {
-        route.infos.produces = ['applications/json'];
+      if (route.info.tags === undefined) {
+        route.info.tags = [];
       }
-
-      if (route.infos.parameters === undefined) {
-        route.infos.parameters = [];
-
-        let m = routeUrlMatch.exec(route.url);
-        while (m !== null) {
-          routeUrlMatch.lastIndex++;
-          route.infos.parameters.push({
-            description: 'TODO',
-            in: 'path',
-            name: m[1],
-            required: true,
-            type: 'string'
-          });
-
-          m = routeUrlMatch.exec(route.url);
-        }
+      if (!route.info.tags.includes(route.controller)) {
+        route.info.tags.push(route.controller);
       }
+    }
 
-      if (route.infos.parameters.length === 0) {
-        route.infos.parameters = undefined;
+    if (route.info.description === undefined) {
+      route.info.description = `Controller: ${route.controller}.`;
+    }
+    if (route.info.summary === undefined) {
+      route.info.summary = `Action: ${route.action}.`;
+    }
+    if (route.info.parameters === undefined) {
+      route.info.parameters = [];
+
+      let m = routeUrlMatch.exec(route.url);
+      while (m !== null) {
+        routeUrlMatch.lastIndex++;
+        route.info.parameters.push({
+          name: m[1],
+          in: 'path',
+          required: true,
+          schema: {type: 'integer'}
```

Comment:
Why integer? Isnt always string in URL?

Reply:
Ops, slipped test. Should be _string_ as default.

```suggestion
          schema: {type: 'string'}
```

Btw, even thought url's contain strings params, the whole idea of having a full api documentation is to be able to _also_ precisely define parameter types, this may help us in many ways (using the api-ui, generating code from this doc..). 
[Schema.type is mandatory in OAS, and it is quite complete (required, defaults, enums, integer ranges...).](https://swagger.io/docs/specification/describing-parameters/)